sdk/src/client/secret/mod.rs

```diff
@@ -115,6 +116,160 @@ pub trait SecretManage: Send + Sync {
+        Ok(self.sign_ed25519(msg, chain).await?)
+    }
+
+    async fn dyn_sign_secp256k1_ecdsa(
+        &self,
+        msg: &[u8],
+        chain: Bip44,
+    ) -> Result<(secp256k1_ecdsa::PublicKey, secp256k1_ecdsa::RecoverableSignature), Error> {
+        Ok(self.sign_secp256k1_ecdsa(msg, chain).await?)
+    }
+
+    async fn dyn_signature_unlock(&self, essence_hash: &[u8; 32], chain: Bip44) -> Result<Unlock, Error> {
+        Ok(self.signature_unlock(essence_hash, chain).await?)
+    }
+
+    async fn dyn_sign_transaction_essence(
+        &self,
+        prepared_transaction_data: &PreparedTransactionData,
+    ) -> Result<Unlocks, Error> {
+        Ok(self.sign_transaction_essence(prepared_transaction_data).await?)
+    }
+
+    async fn dyn_sign_transaction(
+        &self,
+        prepared_transaction_data: PreparedTransactionData,
+    ) -> Result<TransactionPayload, Error> {
+        Ok(self.sign_transaction(prepared_transaction_data).await?)
+    }
+}
+#[async_trait]
+impl SecretManage for dyn DynSecretManagerConfig {
+    type Error = Error;
+
+    async fn generate_ed25519_addresses(
+        &self,
+        coin_type: u32,
+        account_index: u32,
+        address_indexes: Range<u32>,
+        options: impl Into<Option<GenerateAddressOptions>> + Send,
+    ) -> Result<Vec<Ed25519Address>, Self::Error> {
+        self.dyn_generate_ed25519_addresses(coin_type, account_index, address_indexes, options.into())
+            .await
+    }
+
+    async fn generate_evm_addresses(
+        &self,
+        coin_type: u32,
+        account_index: u32,
+        address_indexes: Range<u32>,
+        options: impl Into<Option<GenerateAddressOptions>> + Send,
+    ) -> Result<Vec<EvmAddress>, Self::Error> {
+        self.dyn_generate_evm_addresses(coin_type, account_index, address_indexes, options.into())
+            .await
+    }
+
+    async fn sign_ed25519(&self, msg: &[u8], chain: Bip44) -> Result<Ed25519Signature, Self::Error> {
+        self.dyn_sign_ed25519(msg, chain).await
+    }
+
+    async fn sign_secp256k1_ecdsa(
+        &self,
+        msg: &[u8],
+        chain: Bip44,
+    ) -> Result<(secp256k1_ecdsa::PublicKey, secp256k1_ecdsa::RecoverableSignature), Self::Error> {
+        self.dyn_sign_secp256k1_ecdsa(msg, chain).await
+    }
+
+    async fn signature_unlock(&self, essence_hash: &[u8; 32], chain: Bip44) -> Result<Unlock, Self::Error> {
+        self.dyn_signature_unlock(essence_hash, chain).await
+    }
+
+    async fn sign_transaction_essence(
+        &self,
+        prepared_transaction_data: &PreparedTransactionData,
+    ) -> Result<Unlocks, Self::Error> {
+        self.dyn_sign_transaction_essence(prepared_transaction_data).await
+    }
+
+    async fn sign_transaction(
+        &self,
+        prepared_transaction_data: PreparedTransactionData,
+    ) -> Result<TransactionPayload, Self::Error> {
+        self.dyn_sign_transaction(prepared_transaction_data).await
+    }
```

Comment:
Considering https://github.com/iotaledger/iota-sdk/issues/701, this also looks annoying to implement for other people
Probably worth to check if we can get rid of them and only rely on `sign_ed25519()` if that doesn't make it too complicated and maybe even with `chain: Option<Bip44>`

Reply:
They don't have to implement anything except for `SecretManage`

Reply:
Ok commented on the wrong section then, but there we have the same methods the need to implement, no?

Reply:
Yeah, same as always

Reply:
Yes we have them already, I'm just wondering if it now would make sense to get rid of something
Had a quick look and `sign_transaction()` for example seems to just forward to `default_sign_transaction()` which internally uses `sign_transaction_essence()`, so if I didn't miss something we can just remove that one?
And `signature_unlock()` could also be just a function that accepts a secret manager and internally calls `sign_ed25519()` on it

At least to me it looks like we would have then 2 functions less in the trait and can still do the same things

Reply:
I don't think we can remove either `sign_transaction` or `sign_transaction_essence` just because there are defaults for them. The defaults are internal anyway, not part of the trait. And I don't really understand the point about `signature_unlock` since users don't need to impl that fn anyway.